client/devices/client.go

```diff
@@ -0,0 +1,181 @@
+	ID           string `json:"id"`
+	IdentityData struct {
+		Mac string `json:"mac"`
+		Sku string `json:"sku"`
+		Sn  string `json:"sn"`
+	} `json:"identity_data"`
+	Status    string `json:"status"`
+	CreatedTs string `json:"created_ts"`
+	UpdatedTs string `json:"updated_ts"`
+	AuthSets  []struct {
+		ID           string `json:"id"`
+		PubKey       string `json:"pubkey"`
+		IdentityData struct {
+			Mac string `json:"mac"`
+			Sku string `json:"sku"`
+			Sn  string `json:"sn"`
+		} `json:"identity_data"`
+		Status string `json:"status"`
+		Ts     string `json:"ts"`
+	} `json:"auth_sets"`
+	Decommissioning bool `json:"decommissioning"`
+}
+
+const (
+	devicesListURL = "/api/management/v2/devauth/devices"
+)
+
+type Client struct {
+	url            string
+	devicesListURL string
+	client         *http.Client
+}
+
+func NewClient(url string, skipVerify bool) *Client {
+	tr := &http.Transport{
+		TLSClientConfig: &tls.Config{InsecureSkipVerify: skipVerify},
+	}
+
+	return &Client{
+		url:            url,
+		devicesListURL: JoinURL(url, devicesListURL),
+		client: &http.Client{
+			Transport: tr,
+		},
+	}
+}
+
+func (c *Client) ListDevices(tokenPath string, detailLevel int) error {
+	if detailLevel > 3 || detailLevel < 0 {
+		return fmt.Errorf("FAILURE: invalid devices detail")
+	}
+	token, err := ioutil.ReadFile(tokenPath)
+	if err != nil {
+		return errors.Wrap(err, "Please Login first")
+	}
+
+	req, err := http.NewRequest(http.MethodGet, c.devicesListURL, nil)
+	if err != nil {
+		return errors.Wrap(err, "Failed to create HTTP request")
+	}
+	req.Header.Set("Authorization", "Bearer "+string(token))
+
+	reqDump, err := httputil.DumpRequest(req, false)
+	if err != nil {
+		return err
+	}
+	log.Verbf("sending request: \n%s", string(reqDump))
+
+	rsp, err := c.client.Do(req)
+	if err != nil {
+		return errors.Wrap(err, "Get /devauth/devices request failed")
+	}
+	if rsp.StatusCode != 200 {
+		return fmt.Errorf("Get /devauth/devices request failed with status %d\n", rsp.StatusCode)
+	}
+
+	defer rsp.Body.Close()
+
+	body, err := ioutil.ReadAll(rsp.Body)
+	if err != nil {
+		return err
+	}
+
+	var list devicesList
+	err = json.Unmarshal(body, &list.devices)
+	if err != nil {
+		return err
+	}
+	for _, v := range list.devices {
+		listDevice(v, detailLevel)
+	}
+
+	return nil
+}
+
+func listDevice(a deviceData, detailLevel int) {
+	fmt.Printf("ID: %s\n", a.ID)
+	fmt.Printf("Status: %s\n", a.Status)
+	if detailLevel >= 1 {
+		fmt.Println("IdentityData:")
+		if a.IdentityData.Mac != "" {
+			fmt.Printf("  Mac: %s\n", a.IdentityData.Mac)
+		}
+		if a.IdentityData.Sku != "" {
+			fmt.Printf("  Sku: %s\n", a.IdentityData.Sku)
+		}
+		if a.IdentityData.Sn != "" {
+			fmt.Printf("  SN: %s\n", a.IdentityData.Sn)
+		}
+	}
+	if detailLevel >= 1 {
+		fmt.Printf("CreatedTs: %s\n", a.CreatedTs)
+		fmt.Printf("UpdatedTs: %s\n", a.UpdatedTs)
+		fmt.Printf("Decommissioning: %t\n", a.Decommissioning)
+	}
+	if detailLevel >= 2 {
+		for i, v := range a.AuthSets {
```

Comment:
Feel free to ignore:
Could be a nice use-case for `text-templates` I think

Reply:
Maybe next time... I have to learn them first :)